internal/quotes/quotes_test.go

```diff
@@ -9,14 +9,360 @@ import (
 	"testing"
 )
 
+//				Test - SearchByQuoteTag
+// ====================================================== \\
+
+// TestSearchByQuoteTag tests the SearchByQuoteTag function.
+func TestSearchByQuoteTag(t *testing.T) {
+	// Define a set of sample quotes to use across test cases.
+	sampleQuotes := []Quote{
+		{
+			Text:   "The only way to do great work is to love what you do.",
+			Author: "Steve Jobs",
+			Tags:   []string{"inspiration", "work", "passion"},
+		},
+		{
+			Text:   "Be yourself; everyone else is already taken.",
+			Author: "Oscar Wilde",
+			Tags:   []string{"identity", "humor"},
+		},
+		{
+			Text:   "The future belongs to those who believe in the beauty of their dreams.",
+			Author: "Eleanor Roosevelt",
+			Tags:   []string{"future", "dreams", "inspiration"},
+		},
+		{
+			Text:   "Innovation distinguishes between a leader and a follower.",
+			Author: "Steve Jobs",
+			Tags:   []string{"innovation", "leadership", "work"},
+		},
+		{
+			Text:   "To be or not to be, that is the question.",
+			Author: "William Shakespeare",
+			Tags:   []string{"philosophy", "drama"},
+		},
+	}
+
+	// Define test cases. Each test case has a name, input, and expected output.
+	tests := []struct {
+		name           string
+		quotes         []Quote
+		targetTag      string
+		expectedQuotes []Quote
+		expectedError  error // Placeholder for error handling, though current func doesn't return errors
+	}{
+		{
+			name:      "Find single matching quote",
+			quotes:    sampleQuotes,
+			targetTag: "humor",
+			expectedQuotes: []Quote{
```

Comment:
Would be cleaner if each quote was defined in a variable above. Then
expected quotes: []Quote{be yourself},